packages/react-native/Libraries/Blob/URL.js

```diff
@@ -55,74 +55,125 @@ if (
 // Small subset from whatwg-url: https://github.com/jsdom/whatwg-url/tree/master/src
 // The reference code bloat comes from Unicode issues with URLs, so those won't work here.
 export class URLSearchParams {
-  _searchParams: Array<Array<string>> = [];
+  _params: Array<Array<string>> = [];
 
-  constructor(params: any) {
-    if (typeof params === 'object') {
-      Object.keys(params).forEach(key => this.append(key, params[key]));
+  constructor(init?: string | {[key: string]: string | string[] | void}) {
```

Comment:
is this right? the IDL shows this:

```idl
sequence<sequence<USVString>> or record<USVString, USVString> or USVString
```

which suggests it should be:

```ts
init?: Array<Array<string>> | Record<string, string> | string
```

i suspect the sequence of strings is actually a tuple too, like entries: `Array<[string, string]>`